crates/starknet-server/tests/test_restart.rs

```diff
@@ -0,0 +1,255 @@
+        let predeployed_account = Arc::new(SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer,
+            address,
+            chain_id::TESTNET,
+            ExecutionEncoding::Legacy,
+        ));
+
+        // prepare class for estimation of declaration
+        let contract_json = dummy_cairo_0_contract_class();
+        let contract_artifact: Arc<LegacyContractClass> =
+            Arc::new(serde_json::from_value(contract_json.inner).unwrap());
+
+        // check gas price via fee estimation
+        let estimate_before = predeployed_account
+            .declare_legacy(contract_artifact.clone())
+            .estimate_fee()
+            .await
+            .unwrap();
+        assert_eq!(estimate_before.gas_price, expected_gas_price);
+
+        devnet.restart().await.unwrap();
+
+        let estimate_after =
+            predeployed_account.declare_legacy(contract_artifact).estimate_fee().await.unwrap();
+
+        // assert gas_price and fee are equal to the values before restart
+        assert_eq!(estimate_before.gas_price, estimate_after.gas_price);
+        assert_eq!(estimate_before.overall_fee, estimate_after.overall_fee);
+    }
+
+    #[tokio::test]
+    async fn assert_predeployed_account_is_prefunded_after_restart() {
+        let initial_balance = 1_000_000_u32;
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--initial-balance",
+            &initial_balance.to_string(),
+        ])
+        .await
+        .unwrap();
+
+        let predeployed_account_addresss = devnet.get_first_predeployed_account().await.1;
+
+        let balance_before = devnet.get_balance(&predeployed_account_addresss).await.unwrap();
+        assert_eq!(balance_before, FieldElement::from(initial_balance));
+
+        devnet.restart().await.unwrap();
+
+        let balance_after = devnet.get_balance(&predeployed_account_addresss).await.unwrap();
+        assert_eq!(balance_before, balance_after);
+    }
+
+    #[tokio::test]
+    async fn assert_dumping_not_affected_by_restart() {
+        let dump_file_name = "dump_after_restart";
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--dump-path",
+            dump_file_name,
+            "--dump-on",
+            "exit",
+        ])
+        .await
+        .unwrap();
+
+        devnet.restart().await.unwrap();
+
+        // send a dummy tx; otherwise there's no dump
+        devnet.mint(FieldElement::ONE, 1).await;
+
+        // assert dump file not already here
+        assert!(!Path::new(dump_file_name).exists());
+
+        // assert killing the process can still dump devnet
+        send_ctrl_c_signal(&devnet.process).await;
+        std::thread::sleep(std::time::Duration::from_secs(1)); // sleep to allow dump
+        assert!(Path::new(dump_file_name).exists());
+
+        remove_file(dump_file_name);
+    }
+
+    #[tokio::test]
+    async fn assert_load_not_affecting_restart() {
```

Comment:
We could also add test with dump, restart, load, check if tx is there but this seems to be more dump and load test than restart test. I can also add it by my own during https://github.com/0xSpaceShard/starknet-devnet-rs/issues/237

Reply:
Yes, looks more like a dump/load test, I don't know if we need it.